src/Generator/Generators/CSharp/CSharpMarshal.cs

```diff
@@ -250,7 +250,7 @@ public override bool VisitTypedefType(TypedefType typedef, TypeQualifiers quals)
                 Context.SupportBefore.WriteLine("var {0} = {1};", ptrName,
                     Context.ReturnVarName);
 
-                Context.Return.Write("({1})Marshal.GetDelegateForFunctionPointer({0}, typeof({1}))",
+                Context.Return.Write("{0} == IntPtr.Zero ? null : ({1})Marshal.GetDelegateForFunctionPointer({0}, typeof({1}))",
```

Comment:
Please extract the delegate expression to a separate string and use interpolated strings - we avoid multi-lined ones.